exporter/coralogixexporter/client.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"crypto/tls"
+
+	cxpb "github.com/coralogix/opentelemetry-cx-protobuf-api/coralogixpb"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// CoralogixClient by Coralogix
+type CoralogixClient struct {
+	cfg      Config
+	logger   *zap.Logger
+	conn     *grpc.ClientConn
+	client   cxpb.CollectorServiceClient
+	metadata cxpb.Metadata
+}
+
+// NewCoralogixClient by Coralogix
+func NewCoralogixClient(cfg *Config, logger *zap.Logger) *CoralogixClient {
+	c := &CoralogixClient{
+		cfg:    *cfg,
+		logger: logger,
+		metadata: cxpb.Metadata{
+			ApplicationName: cfg.AppName,
+			SubsystemName:   cfg.SubSystem,
+		},
+	}
+	c.startConnection()
+	return c
+}
+func (c *CoralogixClient) startConnection() {
+	confTLS := &tls.Config{
+		InsecureSkipVerify: false,
+	}
+	conn, err := grpc.Dial(c.cfg.GRPCClientSettings.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(confTLS)))
+	if err != nil {
+		c.logger.Error("connection error cannot dial to the endpoint ", zap.Error(err))
+	}
+	c.conn = conn
+	c.client = cxpb.NewCollectorServiceClient(c.conn)
+}
+
+func (c *CoralogixClient) newPost(ctx context.Context, td pdata.Traces) {
+	batches, err := jaeger.InternalTracesToJaegerProto(td)
+	if err != nil {
+		c.logger.Error("cant translate to jaeger proto", zap.Error(err))
+	}
+	ctx = metadata.NewOutgoingContext(ctx, metadata.New(c.cfg.GRPCClientSettings.Headers))
+	for _, batch := range batches {
+		_, err := c.client.PostSpans(ctx, &cxpb.PostSpansRequest{Batch: *batch, Metadata: &c.metadata}, grpc.WaitForReady(false))
+		if err != nil {
+			c.logger.Error("failed to push trace data via Coralogix exporter ", zap.Error(err))
```

Comment:
If you just return the error and bubble it up, your users can take advantage of the retry logic that can be configured as part of their pipeline. If you don't return the error, the collector will assume this worked and won't retry.

Reply:
Are you going to address this? It's your component, you can decide to not do it, but IMO, it gives your users a poor and inconsistent experience.

Reply:
Thanks, I will fix it, the error will bubble up to the users.